packages/puppeteer-core/src/common/Cookie.ts

```diff
@@ -29,6 +29,23 @@ export type CookiePriority = 'Low' | 'Medium' | 'High';
  */
 export type CookieSourceScheme = 'Unset' | 'NonSecure' | 'Secure';
 
+/**
+ * Represents a cookie partition key in Chrome.
+ *
+ * @public
+ */
+export interface ChromeCookiePartitionKey {
```

Comment:
Let's also define a generic partition key structure as defined by WebDriver BiDi. Currently, it should only contain the sourceOrigin attribute. Let's map sourceOrigin to topLevelSite and vice versa. We should still keep this interface for users only interested in Chrome-specific partition attributes.

Reply:
So, do we want to still support receiving string or only one of the `ChromeCookiePartitionKey` and `BiDiCookiePartitionKey` interfaces (and `undefined` of course)?

Reply:
In any case, I think it might be worthwhile to separate this issue from the current one